src/main/java/com/sourcegraph/webhook/WebhookListener.java

```diff
@@ -0,0 +1,74 @@
+package com.sourcegraph.webhook;
+
+import com.atlassian.bitbucket.event.ApplicationEvent;
+import com.atlassian.bitbucket.event.backup.BackupStartedEvent;
+import com.atlassian.bitbucket.event.permission.PermissionEvent;
+import com.atlassian.bitbucket.event.pull.*;
+import com.atlassian.bitbucket.repository.Repository;
+import com.atlassian.bitbucket.scm.event.ScmEvent;
+import com.atlassian.event.api.AsynchronousPreferred;
+import com.atlassian.event.api.EventListener;
+import com.google.common.base.Joiner;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.sourcegraph.webhook.registry.Webhook;
+import com.sourcegraph.webhook.registry.WebhookRegistry;
+
+import javax.inject.Named;
+import java.util.*;
+
+@AsynchronousPreferred
+@Named("WebhookListener")
+public class WebhookListener {
+    private static Map<Class<?>, List<String>> triggers = new HashMap<>();
+
+    private static void register(Class<?> type, String key) {
+        String[] split = key.split(":");
+        List<String> pieces = new ArrayList<>();
+        for (int i = 0, index = 0; i < split.length; i++) {
+            index += split[i].length();
+            pieces.add(key.substring(0, index));
+            index++;
+        }
+        triggers.put(type, pieces);
+    }
+
+    static {
+        register(PullRequestOpenedEvent.class, "pr:opened");
+        register(PullRequestUpdatedEvent.class, "pr:modified");
```

Comment:
```suggestion
        register(PullRequestUpdatedEvent.class, "pr:updated");
```

Reply:
These keys are from the atlassian docs: [event paylods](https://confluence.atlassian.com/bitbucketserver0516/event-payload-966061436.html?utm_campaign=in-app-help&utm_medium=in-app-help&utm_source=stash#Eventpayload-repositoryevents).

Reply:
I see. Worth documenting that.